src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Backup/BackupStep.cs

```diff
@@ -89,6 +92,11 @@ protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgrad
             }
         }
 
+        private void AddResultToContext(IUpgradeContext context, string backupLocation, UpgradeStepStatus status, string description)
+        {
+            context.AddResult("Backup Step", backupLocation, Id, status, description);
```

Comment:
why not use `.Title` here? Perhaps even add an overload to the new `.AddResult` that takes an `UpgradeStep` obj and does this automatically? Just trying to clean up the code and make usage of this new capability as consistent and affordable as possible

Reply:
Not sure if adding UpgradeStep obj would add much value. Most of the information here cannot be directly deduced from the step object but is local to the particular class or is a local variable. We can only obtain Id and maybe title from the step. But then we have cases like try-convert step where the upgrade is not performed in the step class but a separate class - TryConvertRunner - which doesn't have access to the step object, the Id or the Title properties. 

I believe we might end up over engineering. What do you think?